source/MaterialXRender/Handlers/ImageHandler.h

```diff
@@ -36,6 +38,16 @@ class ImageDesc
     }
 };
 
+/// class @ImagePropertiesDesc
+/// Interface to describe image properties.
+class ImagePropertiesDesc
+{
+  public:
+    MaterialX::ValuePtr uaddressMode;
+    MaterialX::ValuePtr vaddressMode;
+    MaterialX::ValuePtr filterType;
+};
```

Comment:
Maybe name this class just ImageProperties?

Also, for the properties members I think native data types instead of ValuePtr would be better. Then a client can set the properties from anywhere not only a MaterialX document.

Reply:
Ignore the remark above class name above, I reviewed an earlier revision.

Reply:
Yes, I was going to do this but the problem is the type for is different depending on language. For OSL these are strings for GLSL these are integers. I also remap the unmapped color from float,vec2/3/4 and color2/3/4 on the GL side to float4 since that is the input required but may not be for other languages.

Reply:
Ah ok. But is this class not only used for HW? An offline / OSL renderer would have it's own way of binding and sampling textures.

And for the unmapped color it's being converted to a Color4 anyway in mapValueToColor, so I think it could be stored as Color4 directly instead.

Reply:
There is very little usage of MatX here so I've removed even Color4 and just stored float[4]